diesel/src/mysql/connection/url.rs

```diff
@@ -29,8 +29,10 @@ impl ConnectionOptions {
             return Err(connection_url_error());
         }
 
-        let host = match url.host_str() {
-            Some(host) => Some(try!(CString::new(host.as_bytes()))),
+        let host = match url.host() {
+            // Avoid brackets around IPv6 addresses.
+            Some(Host::Ipv6(host)) => Some(try!(CString::new(host.to_string()))),
+            Some(host) => Some(try!(CString::new(host.to_string()))),
```

Comment:
Can this be written as `Some(Host::Ipv6(host)) | Some(host) => Some(CString::new(host.to_string())?),`

Reply:
Unfortunately it can't because `host` has different types in the two match arms (`std::net::Ipv6Addr` vs `mysql::connection::url::url::Host`). 

I can change it to use the `?` operator instead of the `try!` macro -- I only chose to use the latter to maintain consistency with the surrounding code.

Reply:
No, it's fine to leave it using `try!` for now.